test/replaytests/replay_test.go

```diff
@@ -53,6 +53,14 @@ func TestReplayChildWorkflowBugBackport(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// Gives a non-deterministic-error because the getNameActivity3 was not registered on the replayer.
```

Comment:
`getNameActivity3` isn't in the greetings.json history or in the workflow code.  what's this testing?

Reply:
updated the documentation. this is testing the scenario where we get a non deterministic error if we don't register the activities in the replay test code.